farmfs/snapshot.py

```diff
@@ -130,7 +152,7 @@ def __str__(self):
       return "{"+self.path("")+","+self.mode+","+self.csum+"}" # Not a great encoding.
 
 def encode_snapshot(snap):
-  return map(lambda x: x.get_dict(), snap)
+  return list(map(lambda x: x.get_dict(), snap))
```

Comment:
should be using imap, and maybe return the iterator so its easier to convert to lazy encoder.